sylvia/tests/api.rs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        #[msg(exec)]
+        pub fn exec(&self, _ctx: ExecCtx, _param: Exec) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[msg(query)]
+        pub fn query(&self, _ctx: QueryCtx, _param: Query) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[msg(migrate)]
+        pub fn migrate(&self, _ctx: MigrateCtx, _param: Migrate) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::some_contract::sv::InstantiateMsg;
+    use crate::some_contract::SomeContract;
+    use cosmwasm_std::Addr;
+    use sylvia::types::{ContractApi, SvCustomMsg};
+
+    #[test]
+    fn api() {
+        let owner = Addr::unchecked("owner");
+
+        let _: InstantiateMsg<SvCustomMsg> = <SomeContract<
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+        > as ContractApi>::Instantiate::new(
+            SvCustomMsg
+        );
+
+        let exec: crate::some_contract::sv::ExecMsg<SvCustomMsg> = <SomeContract<
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+        > as ContractApi>::Exec::exec(
+            SvCustomMsg
+        );
+
+        let query: crate::some_contract::sv::QueryMsg<SvCustomMsg> = <SomeContract<
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+        > as ContractApi>::Query::query(
+            SvCustomMsg
+        );
+
+        let _: crate::some_contract::sv::ContractExecMsg<SvCustomMsg> = <SomeContract<
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+        > as ContractApi>::ContractExec::SomeContract(
+            exec
+        );
+
+        let _: crate::some_contract::sv::ContractQueryMsg<SvCustomMsg> = <SomeContract<
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+        > as ContractApi>::ContractQuery::SomeContract(
+            query
+        );
+
+        let _: crate::some_contract::sv::Remote<'_> = <SomeContract<
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+            SvCustomMsg,
+        > as ContractApi>::Remote::new(owner);
+
+        // let deps = mock_dependencies();
+        // let querier_wrapper: QuerierWrapper = QuerierWrapper::new(&deps.querier);
+        // let _: crate::some_contract::sv::BoundQuerier<'_, cosmwasm_std::Empty> = <SomeContract<
+        //     SvCustomMsg,
+        //     SvCustomMsg,
+        //     SvCustomMsg,
+        //     SvCustomMsg,
+        //     SvCustomMsg,
+        // > as ContractApi>::Querier::borrowed(
+        //     &owner,
+        //     &querier_wrapper,
+        // );
```

Comment:
Not sure why compiler expects `querier_wrapper` and `owner` to be `'static` in this test.